src/main/java/com/example/Restaurant_Management_System_REST_API/model/ContactDetails.java

```diff
@@ -0,0 +1,18 @@
+package com.example.Restaurant_Management_System_REST_API.model;
+
+import jakarta.persistence.MappedSuperclass;
+import lombok.Getter;
+import lombok.Setter;
+
+@Getter
+@Setter
+@MappedSuperclass
+public abstract class ContactDetails {
+
+    private String name;
+    private String street;
+    private String city;
+    private int postalCode;
+    private int telephoneNumber;
+    private String emailAddress;
```

Comment:
Are those fields enough for an address?

Reply:
I think so. Are you proposing anything else?

Reply:
I won't suggest anything more at this stage, as I want you to solve it and think :)